java-spring/security/jv-spring-security-checklist.md

```diff
@@ -11,7 +11,7 @@
 1. Remember, that `null.equals(smth)` == NPE. HINT: be careful with implementation of the `isValid()` method.
 1. Don't repeat checks for fields: you must ensure `email` field not to be null in your custom Validator, so there is no need to put `@NotNull` for that field as well in Dto class.
 1. Can we use `@NotNull` annotation above the field with primitive data types? Be careful to use only the suitable annotations :)
-1. Use `@Autowired` with `PasswordEncoder` in `UserServiceImpl`.
+1. Use `@Autowired` with `PasswordEncoder` in `UserServiceImpl` if you don't use the constructor.
```

Comment:
better something like this:
1. Don't create PasswordEncoder using `new` keyword. Let's use dependency injection by the constructor.